src/Extensions/TaxonomyTermUrlExtension.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace SilverStripe\Taxonomy\Extensions;
+
+use SilverStripe\Taxonomy\TaxonomyTerm;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\Forms\FieldList;
+
+/**
+* Adds a URLSegment to TaxonomyTerm that gets
+* generated from the name if not specified.
+*
+* @package taxonomy
+*/
+class TaxonomyTermUrlExtension extends DataExtension
+{
+    private static $db = array(
+        'URLSegment' => 'Varchar(255)',
+    );
+
+    public function updateCMSFields(FieldList $fields)
+    {
+        $field = $fields->dataFieldByName('URLSegment');
+        if ($field) {
+            $field->setDescription('Leave blank to generate from name');
+            $fields->insertAfter('Name', $field);
+        }
+    }
+
+    /**
+     *  Set the URL segment allowing filtering by url slug
+     *
+     *  {@inheritDoc}
+     */
+    public function onBeforeWrite() {
```

Comment:
If you could run a phpcs linter over this, that's be great

Reply:
It's no longer being picked up, but there's still inconsistent spacing in this method. Would you mind adjusting it manually?